oss-internship-2020/sapi_lodepng/examples/main_unit_test.cc

```diff
@@ -0,0 +1,225 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <filesystem>
+
+#include "gtest/gtest.h"
+#include "sandbox.h"
+#include "sandboxed_api/util/status_matchers.h"
+
+using sapi::IsOk;
+using testing::Eq;
+using testing::NotNull;
+
+namespace {
+
+// use the current path
+const std::string images_path = std::filesystem::current_path().string();
+
+TEST(initSandbox, basic) {
+  SapiLodepngSandbox sandbox(images_path);
+  ASSERT_THAT(sandbox.Init(), IsOk()) << "Error during sandbox init";
+}
+
+// generate an image, encode it, decode it and compare the pixels with the
+// initial values
+TEST(generate_image, encode_decode_compare_one_step) {
+  SapiLodepngSandbox sandbox(images_path);
+  ASSERT_THAT(sandbox.Init(), IsOk()) << "Error during sandbox init";
+  LodepngApi api(&sandbox);
+
+  // generate the values
+  constexpr unsigned int width = 512, height = 512,
```

Comment:
maybe extract this generation code into a function.
It's repeated at least 4 times.